site-src/geps/gep-992.md

```diff
@@ -0,0 +1,184 @@
+# GEP-992: HTTPRoute Load-Balancing Strategies
+
+* Issue: [#992](https://github.com/kubernetes-sigs/gateway-api/issues/992)
+* Status: Provisional
+
+## TLDR
+
+Explicit and named strategies for load-balancing multiple `HTTPBackendRef`
+
+## Goals
+
+- make configuring a load-balancing strategy explicit and provide defaults
+- add the "least-connections" strategy
+
+## Non-Goals
+
+- defining combinations of strategies (e.g. "least-connections" + "weighted")
+
+## Introduction
+
+It's currently possible to configure multiple backends for an `HTTPRoute` rule, e.g.:
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1alpha2
+kind: HTTPRoute
+metadata:
+  name: multiple-backends
+spec:
+  rules:
+  - matches:
+    - path:
+        type: PathPrefix
+        value: /bar
+    backendRefs:
+    - name: service1
+      port: 80
+    - name: service2
+      port: 80
+    - name: service3
+      port: 80
+```
+
+This implies a load-balancing strategy for traffic which matches but is limited
+in how it can be configured today.
+
+### Problems
+
+Currently only one load-balancing strategy is available, which we'll call the
+`weighted` strategy (and by proxy, you can implement a `round-robin` strategy
```

Comment:
We should be careful as round robin is distinct from weighted. For example, flipping a coin for each request is ok for the weighted implementation, but round robin is more deterministic in behavior.

Reply:
Correct, thus why in the implementation hereafter they are referred to as distinct items and have their own distinct types.